packages/core/src/amazonq/util/telemetryHelper.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MetricName } from '../../shared/telemetry/clienttelemetry'
+
+// TODO: technically this can also be any metric name that occurs in the flow now
+type Event = 'initialRequestTime' | 'webviewToDispatcher' | 'dispatcherToFeature' | 'finalRequestTime' | MetricName
+
+export class AmazonQUITelemetryHelper {
+    static #instance: AmazonQUITelemetryHelper
+
+    static get instance(): AmazonQUITelemetryHelper {
+        return (this.#instance ??= new AmazonQUITelemetryHelper())
+    }
+
+    private UIMetrics = new Map<
+        string,
+        [
+            {
+                event: Event
+                time: number
+            },
+        ]
+    >()
+
+    /**
+     * Record the time an event happened in a tabID or conversation
+     * @param tabIDOrConversationId an amazon Q tabID or a conversation ID
+     * @param event a UI or telemetry event
+     * @param time the duration it took
```

Comment:
I'm seeing uses where it looks like this value being set is a point in time and not the actual duration/diff. Does this need to be updated to something like `startTime`?